packages/cheqd-sdk/src/dids/CheqdDidRegistrar.ts

```diff
@@ -0,0 +1,425 @@
+    const didRepository = agentContext.dependencyManager.resolve(DidRepository)
+    const cheqdLedgerService = agentContext.dependencyManager.resolve(CheqdLedgerService)
+    const { didDocument, didDocumentMetadata } = await cheqdLedgerService.resolve(did)
+    const didRecord = await didRepository.findCreatedDid(agentContext, did)
+    if (!didDocument || didDocumentMetadata.deactivated || !didRecord) {
+      return {
+        resourceMetadata: {},
+        resourceRegistrationMetadata: {},
+        resourceState: {
+          state: 'failed',
+          reason: `DID: ${did} not found`,
+        },
+      }
+    }
+
+    try {
+      let data: Uint8Array
+      if (typeof resource.data === 'string') {
+        data = TypedArrayEncoder.fromBase64(resource.data)
+      } else if (typeof resource.data == 'object') {
+        data = TypedArrayEncoder.fromString(JSON.stringify(resource.data))
+      } else {
+        data = resource.data
+      }
+
+      const resourcePayload = MsgCreateResourcePayload.fromPartial({
+        collectionId: did.split(':')[3],
+        id: resource.id,
+        resourceType: resource.resourceType,
+        name: resource.name,
+        version: resource.version,
+        alsoKnownAs: resource.alsoKnownAs,
+        data,
+      })
+      const payloadToSign = MsgCreateResourcePayload.encode(resourcePayload).finish()
+      const signInputs = await this.signPayload(agentContext, payloadToSign, didDocument.verificationMethod)
+      const response = await cheqdLedgerService.createResource(did, resourcePayload, signInputs)
+      if (response.code !== 0) {
+        throw new Error(`${response.rawLog}`)
+      }
+
+      // TODO: add resource to didRecord
+      // await didRepository.save(agentContext, didRecord)
+
+      return {
+        resourceMetadata: {},
+        resourceRegistrationMetadata: {},
+        resourceState: {
+          state: 'finished',
+          resourceId: resourcePayload.id,
+          resource: resourcePayload,
+        },
+      }
+    } catch (error) {
+      return {
+        resourceMetadata: {},
+        resourceRegistrationMetadata: {},
+        resourceState: {
+          state: 'failed',
+          reason: `unknownError: ${error.message}`,
+        },
+      }
+    }
+  }
+
+  private async signPayload(
+    agentContext: AgentContext,
+    payload: Uint8Array,
+    verificationMethod: VerificationMethod[] = []
+  ) {
+    return await Promise.all(
+      verificationMethod.map(async (method) => {
+        let key: Key
+        if (method.publicKeyBase58) {
+          key = Key.fromPublicKeyBase58(method.publicKeyBase58, KeyType.Ed25519)
+        } else if (method.publicKeyJwk) {
+          key = Key.fromJwk(method.publicKeyJwk as unknown as Jwk)
+        } else if (method.publicKeyMultibase) {
+          key = Key.fromFingerprint(method.publicKeyMultibase)
+        } else {
+          throw new Error(`Invalid verificationMethod`)
+        }
```

Comment:
we have a `getKeyFromVerificationMethod` method (you need to update with latest main). I think we could let this be handled by this method. Then it will also automatically support multiple key types.